packages/grid/x-data-grid-pro/src/tests/treeData.DataGridPro.test.tsx

```diff
@@ -702,4 +702,17 @@ describe('<DataGridPro /> - Tree Data', () => {
       ]);
     });
   });
+
+  describe('regressions', () => {
+    it('should not fail with checkboxSelection', () => {
```

Comment:
```suggestion
    // See https://github.com/mui/mui-x/issues/9402
    it('should not fail with checkboxSelection', () => {
```

For additional context